### DISCUSSION TITLE
Unresolved function or method findUnique() in webstorm

### DISCUSSION THREAD

**Discussion by tenVitality** [prisma/prisma#21492] [Q&A] [answered]
### Bug description

there are some warnings in webstorm, but vscode is normal.

<img width="717" alt="image" src="https://github.com/prisma/prisma/assets/46551420/f1ba1418-d547-40f7-9ad7-b403286f92e7">

### How to reproduce

1. git pull https://github.com/prisma/prisma-examples
2. cd prisma-examples/typescript/rest-nestjs 
3. pnpm i
4. pnpx prisma migrate dev --name init
6. you can see `Unresolved function or method findUnique()` in typescript/rest-nestjs/src/app.controller.ts

### Expected behavior

This may be a configuration problem about pnpm, how should I fix it?

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: Mac m1
- Database: PostgreSQL 
- Node.js version: 18.16.1

I use pnpm to install prisma.

### Prisma Version

```
```

**Comment by youwei997** [comment]
I also encountered the same problem

**Comment by LaicZhang** [comment, marked as answer]
After I use yarn instead of pnpm, everything is ok.

**Reply by max-lf** [reply]
Can confirm that installing via `npm` instead of `pnpm` worked for me as well. On an M1 Macbook Pro.

**Comment by stephanvane** [comment]
Same problem here (pnpm, Mac M2). Some extra information:

1. This warning only shows up for 'finder' methods: (e.g. `findFirstOrThrow`, `findMany`, `upsert`, `updateMany` etc.). Other methods like `count`, `create`, `delete` etc. are working as expected.
2. I'm using a MacBook with a M2 max chip. **The same codebase on a Intel based MacBook works perfectly.** It seems as the warnings are related to the chips's architecture (arm64 vs x64)

I hope this helps narrowing this issue down!

**Comment by bakerEHG** [comment]
Same here

![image](https://github.com/prisma/prisma/assets/132312882/3fe64d8d-7de1-4bc9-bbd1-e912e008e096)

**Reply by Jolg42** [reply]
@bakerEHG Try to Ctrl+Click on `PrismaClient`, because, I tried, and when the `.d.ts` file is open, then WebStorm picks up changes instantly.

**Reply by bakerEHG** [reply]
No luck :(

**Comment by lewsmith** [comment]
I'm pretty sure this is due to project file exclusions. Due to the way Prisma places it's generated code into the `node_modules` directory, your client code is being excluded by IntelliJ by default.

Easy fix, find your client code in the project file list, usually `node_modules/@prisma/client`, right click, 'Mark directory as' and then choose either 'Cancel exclusion' or 'Not Excluded'

Issue should be gone after IntelliJ runs it's file indexer or a restart.

**Reply by Jolg42** [reply]
Thanks, @lewsmith, I wanted to try this, but somehow it didn't work on WebStorm for me. Though it works when I restart WebStorm, I don't think it's a good option, at least for myself.

What works for me:
- Ctrl+Click on `PrismaClient` did refresh the types (that works, like in VS Code)
- or checking `Recompile on changes` in TypeScript settings and then when editing the TS file the types are refreshed, but that’s probably bad DX for most, because that would be for any change.
<img width="993" alt="Screenshot 2023-10-31 at 18 29 53" src="https://github.com/prisma/prisma/assets/1328733/a9277e96-ced1-4dee-a480-bdfa47be715d">

**Reply by Jolg42** [reply]
Note that after doing a Ctrl+Click on `PrismaClient`, it opens the `.d.ts`, and as along as I keep it open, then WebStorm picks up changes instantly.

**Reply by hihabib** [reply]
not working

**Comment by stephanvane** [comment]
hi, @lewsmith 
thanks for the suggestion. Unfortunately this didn't change anything in my case.

As you can see in @bakerEHG's [screenshot](https://github.com/prisma/prisma/discussions/21492#discussioncomment-7399572), the editor clearly knows the method and its types, and also shows documentation for that method. I can't figure out what is triggering the warning.

**Reply by oxwazz** [reply]
@stephanvane, try **un-exclude**  the `node_module` folder and change it to **resource root**, in my case its works perfectly

![image](https://github.com/prisma/prisma/assets/58234878/0aad61bc-6098-411d-bc27-061fbd13f7fe)

**Reply by esainty** [reply]
Note that if using pnpm this needs to be the one in @prisma+client@5.x.x inside the .pnpm folder, not the root node_modules directory. Marking that folder as resource root fixed the issue for me.

**Comment by bakerEHG** [comment]
This seems to be only present in Prisma 5, I have another project with Prisma 4, no issues there

**Comment by Oleksandr-Yemelianov** [comment]
What worked for me was removing node_modules, prisma/client, restart the IDE and install it again

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
